qiskit/providers/aer/backends/aer_compiler.py

```diff
@@ -77,6 +80,28 @@ def compile(self, circuits, basis_gates=None, optypes=None):
             return circuits
         return circuits, optypes
 
+    def _inline_initialize(self, circ, basis_gates):
+        """inline initialize.definition gates if statevector is not used"""
+        need_inline = False
+        for inst, _, _ in circ.data:
+            if isinstance(inst, Initialize) and not isinstance(inst.params[0], complex):
+                need_inline = True
+                break
+
+        if not need_inline:
+            return circ
+
+        new_circ = circ.copy()
+        new_circ.data = []
+        for inst, qargs, cargs in circ.data:
+            if isinstance(inst, Initialize) and not isinstance(inst.params[0], complex):
+                new_circ.compose(transpile(inst.definition, basis_gates=basis_gates),
+                                 qargs, cargs, inplace=True)
```

Comment:
> The call to `transpile` looks awkward here, because it's not going to use the same settings that the user originally used to create the circuit, and it's an expensive call.
> I wonder if it might be better here to actually ignore Terra's definitions, and just use `Initialize.params[0]` directly to generate the statevector. The choices are `Statevector`, `list[complex]`, `str` and `int`, where Aer should natively support the first two already, and the second two can easily be made into an array.

This sounds very good.

> I suppose it'd have different noise characteristics, which is tricky. Perhaps a better approach is for us to have Aer _not_ include `initialize` in its basis gates and require users to do `set_statevector` if they want to losslessly seed the simulator; it seems like a bit of a disservice if we don't support `initialize` the way an actual backend would with regard to noise.

Probably, taking `initialize` has impacts on applications. It may be better to declare no-support of initialization from labels.